core/internals.h

```diff
@@ -288,6 +320,32 @@ typedef enum
     LWM2M_REQUEST_TYPE_DELETE_ALL
 } lwm2m_request_type_t;
 
+#ifdef LWM2M_SUPPORT_SENML_CBOR
+typedef enum {
```

Comment:
Maybe referencing a relevant RFC or the like?

Reply:
CBOR is specified in an Internet Standard RFC. https://www.rfc-editor.org/rfc/rfc8949.html

Reply:
Having a link to this would be really nice anway. But maybe from within `data/cbor.c`?

Reply:
I'm not sure whether the code is generally complaint with rfc8949 or [rfc7049](https://datatracker.ietf.org/doc/html/rfc7049).

Reply:
OMA-TS-LightweightM2M_Core-V1_2-20201110-A references [rfc8428](https://tools.ietf.org/html/rfc8428.txt) for SenML JSON and CBOR.

Reply:
Sorry, If missunderstood. 

But since LWM2M v1.1 there is 2 content format which are using CBOR.


Data Format | IANA Media Type | Numeric Content-Formats [CoAP]
-- | -- | --
CBOR | application/cbor | 60
SenML CBOR | application/senml+cbor | 112

(source : [LWM2M-v1.1.1@core§Table: 7.4.-2 Additional standardized Media Types supported in LwM2M TS 1.1](https://www.openmobilealliance.org/release/LightweightM2M/V1_1_1-20190617-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1_1-20190617-A.html#Table-74-2-Additional-standardized-Media-Types-supported-in-LwM2M-TS-11))

**Usage:**
- First one is only used for singular resource (and resource instance) like Text format.
- Second one is used for everything. (like old JSON format)

**RFC:**
- First one is based on [RFC 7049.](https://www.rfc-editor.org/rfc/rfc7049.html) in LWM2M v1.1.1 but this RFC was obsoleted by [RFC 8949](https://www.rfc-editor.org/rfc/rfc8949). LWM2M v1.2.1 refers RFC 8949. In LWM2M v1.1.1 there very few information about this format, in LWM2M v1.2.1 this is a bit clearer.

- Second one is based on [RFC 8428](https://www.rfc-editor.org/rfc/rfc8428) especially [§6 - CBOR Representation (application/senml+cbor)](https://www.rfc-editor.org/rfc/rfc8428#section-6)  (:point_up: which is based on RFC 7049 which is obsoleted by RFC 8949 :sweat_smile:)

**About RFC 7049  obsoleted by RFC 8949**

RFC 8949 says : 
> This document obsoletes RFC 7049, providing editorial improvements, new details, and errata fixes while keeping full compatibility with the interchange format of RFC 7049. It does not create a new version of the format.

So I guess it is worth to target RFC 8949 even for LWM2M v1.1.1 and SENML-CBOR.